internal/jobservice/application.go

```diff
@@ -100,46 +127,35 @@ func (a *App) StartUp(ctx context.Context, config *configuration.JobServiceConfi
 	}
 
 	// Start a pool
-	pool, err := grpcpool.NewWithContext(ctx, connFactory,
+	evConnPool, err := grpcpool.NewWithContext(ctx, connFactory,
 		config.GrpcPool.InitialConnections,
 		config.GrpcPool.Capacity,
 		0)
 	if err != nil {
 		return err
 	}
 
-	// This function runs in the background every 30 seconds
 	// We will loop over the subscribed jobsets
-	// And we check if we have already subscribed via subscribeMap
 	// If we have then we skip that jobset
 	g.Go(func() error {
+		evClient := events.NewPooledEventClient(evConnPool)
+		subRequests := make(chan SubRequest)
 		ticker := time.NewTicker(30 * time.Second)
-		eventClient := events.NewPooledEventClient(pool)
-		var subscribeMap sync.Map
-		for range ticker.C {
 
-			jobSets, err := sqlJobRepo.GetSubscribedJobSets(ctx)
-			log.Infof("job service has %d subscribed job sets", len(jobSets))
+		for w := 1; w <= config.SubscribeJobSetPoolSize; w++ {
```

Comment:
It's probably just me, but I kept reading this name as a verb -- "Subscribe!" -- but in reality is a thing -- "Pool". Would a name like "SubscriberPool" or something make that more clear? "JobSetObserverPool"?

Reply:
Good catch - I've renamed it to `SubscriberPoolSize` in commit ae717b13.